packages/@aws-cdk/aws-stepfunctions-tasks/package.json

```diff
@@ -179,7 +179,31 @@
       "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.TransformOutput.encryptionKey",
       "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.TransformResources.volumeKmsKeyId",
       "docs-public-apis:@aws-cdk/aws-stepfunctions-tasks.VpcConfig",
-      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.VpcConfig.subnets"
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.VpcConfig.subnets",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrAddStepProps.mainClass",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrAddStepProps.args",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrAddStepProps.properties",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.additionalInfo",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.applications",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.autoScalingRole",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.bootstrapActions",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.configurations",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.customAmiId",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.ebsRootVolumeSize",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.kerberosAttributes",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.logUri",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.releaseLabel",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.scaleDownBehavior",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.securityConfiguration",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.tags",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateClusterProps.visibleToAllUsers",
+      "props-default-doc:@aws-cdk/aws-stepfunctions-tasks.EmrCreateCluster.*",
+      "docs-public-apis:@aws-cdk/aws-stepfunctions-tasks.EmrCreateCluster.*",
+      "duration-prop-type:@aws-cdk/aws-stepfunctions-tasks.EmrCreateCluster.CloudWatchAlarmDefinitionProperty.period",
```

Comment:
Periods should definitely be modeled using `core.Duration`.

Reply:
I originally fixed these linter messages and made them Durations. I also made the SubnetId and SubnetIds L2s. 

But by using cdk value types these values can only be set at deployment time and not at State Machine execution time. By using simple data types we can set the values using `sfn.Data.stringAt()` or`sfn.Data.numberAt()` to dynamically set the values during State Machine execution.

Reply:
You should be able to do 

```ts
Duration.seconds(sfn.Data.numberAt('$.timeout'))
```

As long as no conversion needs to take place, you can pass through lazy values that way.

Reply:
Periods and Timeouts converted to cdk.Durations